vsintegration/src/FSharp.Editor/Refactor/RemoveExplicitReturnType.fs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Threading
+open System.Threading.Tasks
+
+open FSharp.Compiler
+open FSharp.Compiler.CodeAnalysis
+open FSharp.Compiler.Symbols
+open FSharp.Compiler.Text
+open FSharp.Compiler.Syntax
+
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeRefactorings
+open Microsoft.CodeAnalysis.CodeActions
+open CancellableTasks
+open System.Diagnostics
+open Microsoft.CodeAnalysis.Text
+open InternalOptionBuilder
+
+[<ExportCodeRefactoringProvider(FSharpConstants.FSharpLanguageName, Name = "RemoveExplicitReturnType"); Shared>]
+type internal RemoveExplicitReturnType [<ImportingConstructor>] () =
+    inherit CodeRefactoringProvider()
+
+    static member RangeOfReturnTypeDefinition(input: ParsedInput, symbolUseStart: pos, ?skipLambdas) =
+        let skipLambdas = defaultArg skipLambdas true
+
+        SyntaxTraversal.Traverse(
+            symbolUseStart,
+            input,
+            { new SyntaxVisitorBase<_>() with
```

Comment:
To separate the concerns a bit, we tend to put these to the `FSharpParseFileResults` module, it has a lot of similar stuff.

Reply:
Done